gnucash/import-export/import-main-matcher.c

```diff
@@ -361,54 +377,240 @@ gnc_gen_trans_update_toggled_cb (GtkCellRendererToggle *cell_renderer,
+                                GtkTreeViewColumn *column,
+                                GNCImportMainMatcher *info)
+{
+    Account *assigned_account;
+    gboolean first, is_selection;
+
+    ENTER("");
+    assigned_account = NULL;
+    first = TRUE;
+    is_selection = FALSE;
+    gnc_gen_trans_assign_transfer_account(treeview, 
+                            &first,  &is_selection, path,
+                            &assigned_account, info);
+    DEBUG("account returned = %s", gnc_account_get_full_name(assigned_account));
+    LEAVE("");
 }
 
 static void
 gnc_gen_trans_row_changed_cb (GtkTreeSelection *selection,
-                              GNCImportMainMatcher *gui)
+                              GNCImportMainMatcher *info)
 {
     GtkTreeModel *model;
     GtkTreeIter iter;
+    GtkSelectionMode mode;
 
-    if (!gtk_tree_selection_get_selected(selection, &model, &iter))
-        return;
-    gtk_tree_selection_unselect_iter(selection, &iter);
+    ENTER("");
+    mode = gtk_tree_selection_get_mode(selection);
+    switch (mode)
+    {
+        case GTK_SELECTION_MULTIPLE:
+            DEBUG("mode = GTK_SELECTION_MULTIPLE, no action");
+            break;
+        case GTK_SELECTION_NONE:
+            DEBUG("mode = GTK_SELECTION_NONE, no action");
+            break;
+        case GTK_SELECTION_BROWSE:
+            DEBUG("mode = GTK_SELECTION_BROWSE->default");
+        case GTK_SELECTION_SINGLE:
+            DEBUG("mode = GTK_SELECTION_SINGLE->default");
+        default:
+            DEBUG("mode = default unselect selected row");
+            if (gtk_tree_selection_get_selected(selection, &model, &iter))
+            {
+                gtk_tree_selection_unselect_iter(selection, &iter);
+            }
+    }
+    LEAVE("");
+}
+
+static void
+gnc_gen_trans_view_popup_menu (GtkTreeView *treeview, 
+                               GdkEvent *event,
+                               GNCImportMainMatcher *info)
+{    
+    GtkWidget *menu, *menuitem;
+    GdkEventButton *event_button;
+
+    ENTER ("");
+    menu = gtk_menu_new();
+    menuitem = gtk_menu_item_new_with_label(_("Assign a transfer account."));
+    g_signal_connect(menuitem, "activate",
+                     G_CALLBACK(
+                     gnc_gen_trans_assign_transfer_account_to_selection_cb),
+                     info);
+    DEBUG("Callback to assign destination account to selection connected");
+    gtk_menu_shell_append(GTK_MENU_SHELL(menu), menuitem);
+    gtk_widget_show_all(menu);
+    event_button = (GdkEventButton *) event;
+    /* Note: event can be NULL here when called from view_onPopupMenu;
+     *  gdk_event_get_time() accepts a NULL argument */
+    /* replace deprecated function
+    gtk_menu_popup(GTK_MENU(menu), NULL, NULL, NULL, NULL, 
+                    (event_button != NULL) ? event_button->button : 0, 
+                    gdk_event_get_time((GdkEvent*)event)); */
+    gtk_menu_popup_at_pointer(GTK_MENU(menu),(GdkEvent*)event);            
```

Comment:
CI is failing on Ubuntu 14.04 because this function didn't appear until gtk 3.22. You need to leave the call to gtk_menu_popup and guard it like this:
```
#if ! GTK_MAJOR_VERSION < 3 || GTK_MAJOR_VERSION == 3 && GTK_MINOR_VERSION < 22
   gtk_menu_popup(GTK_MENU(menu), NULL, NULL, NULL, NULL, 
                    (event_button != NULL) ? event_button->button : 0, 
                    gdk_event_get_time((GdkEvent*)event));
#else
    gtk_menu_popup_at_pointer(GTK_MENU(menu),(GdkEvent*)event);            
#endif
 ```   
This should be a separate commit called something like "Silence deprecation warning". As a general rule don't comment out old code, just replace it. Old code in comments is just noise.

Reply:
Thanks John 
I'll add that and fix up the problems Bob identified above and recommit it 

David